common/parser.hpp

```diff
@@ -194,50 +196,23 @@ struct commands_xml
     std::map<std::string, custom_formatter> custom_formatters;
 };
 
-inline std::string hexify(unsigned char n)
+// While working from files, convert the file content into recognized raw format
+inline void preprocess_data_payload(std::vector<std::string>& params)
 {
-    std::string res;
-
-    do
+    if (!params.size())
+        return;
+ 
+    //The last parameter provided by the user can represent a fully-qualified file path
+    std::ifstream file(params.back(), std::ios::binary | std::ios::in);
+    if (file.good())
     {
-        res += "0123456789ABCDEF"[n % 16];
-        n >>= 4;
-    } while (n);
-
-    std::reverse(res.begin(), res.end());
+        auto data = std::vector<uint8_t>((std::istreambuf_iterator<char>(file)),
+            std::istreambuf_iterator<char>());
 
-    if (res.size() == 1)
-    {
-        res.insert(0, "0");
+        std::vector<std::string> modified_params(params.size() - 1 + data.size());
```

Comment:
It's still not copying the first `params.size()-1` strings...

Reply:
done